rules/do-you-use-these-useful-react-hooks/rule.md

```diff
@@ -1,161 +1,265 @@
+:::
+
+### ✅Recommended Usage
+
+- **External System Connection**: Link React components to other systems like APIs, networks, or third-party libraries.
+- **Custom Hooks Encapsulation**: Nest your effect logic inside custom hooks for clarity and better structure.
+- **Non-React Widget Control**: Bridge the gap between React components and non-React widgets.
+- **Data Fetching**: While `useEffect` can fetch data, it's optimal to use the framework's standard mechanisms or custom hooks.
+- **Reactive Dependencies**: Identify reactive items (e.g., props, state) that influence your effect. When these alter, the effect kicks in again.
+- **State Updates**: Adjust state values referencing their former versions using `useEffect`.
+- **Access to Recent Props & State**: `useEffect` ensures the most recent props and state are at your disposal.
+- **Distinct Server/Client Content**: With effects operational solely on the client, orchestrate unique content for server and client views.
+
+### ⚠️Pitfalls
+
+- **Placement**: Ensure `useEffect` remains at the top level of your components/custom hooks. Bypass calling it within loops or conditionals.
+- **Avoid Overuse**: Turn to `useEffect` mainly for external synchronization.
+- **Strict Mode Nuances**: In strict mode, a preliminary setup+cleanup cycle, exclusive to development, verifies your cleanup's alignment with your setup.
+- **Dependencies Oversight**: If dependencies comprise inner-component objects or functions, they might trigger frequent effect repetitions.
+- **Visual Effects Caution**: A visual glitch before an effect suggests `useLayoutEffect` might be a better pick.
+- **Server Rendering**: `useEffect` is client-centric and doesn't engage during server-side rendering.
+
+Read more about `useEffect` on the [offical docs.](https://react.dev/reference/react/useEffect)
+
+## 3️⃣ **useContext**: Using Context Seamlessly 🌍
+
+`useContext` is a pivotal React Hook, giving you the power to both read and subscribe to context values right within your component. Below is a breakdown based on the details you provided:
+
+```jsx
+import { createContext, useContext } from 'react';
+
+// Create a context
+const ThemeContext = createContext({
+  background: 'light',
+  foreground: 'dark',
+});
+
+function ThemedButton() {
+  const theme = useContext(ThemeContext);
+  return (
+    <button style={{ background: theme.background, color: theme.foreground }}>
+      I am styled by theme context!
+    </button>
+  );
+}
+
+export default function App() {
+  return (
+    <ThemeContext.Provider value={{ background: 'black', foreground: 'white' }}>
+      <ThemedButton />
+    </ThemeContext.Provider>
+  );
+}
+```
+**Figure: A Themed button example using useContext**
+
+
+### ✅Recommended Usage
+
+- **Reading and Subscribing to Context**: Directly access and subscribe to context values straight from your component.
```

Comment:
1. Please remove full stops after list items

https://www.ssw.com.au/rules/avoid-full-stops-in-bullet-point-lists